ktor-features/ktor-metrics-micrometer-prometheus/jvm/src/io/ktor/metrics/micrometer/prometheus/installFeatureWithAuth.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.metrics.micrometer.prometheus
+
+import io.ktor.application.*
+import io.ktor.auth.*
+import io.ktor.routing.*
+
+fun Application.installMetricsEndpointWithAuth(config: MicrometerMetricsPrometheus.Configuration) {
```

Comment:
We usually call such functions in declarative manner. For example, `routing { }` instead of `installRouting`, `authentication` instead of `installAuthentication` and so on. Also it makes sense to have micrometer in the name. Possible names would be `prometheusEndpoint` or `micrometerPrometheusEndpoint`. 

The other problem is that it requires `confi`g as a parameter while we are usually providing a lambda block to configure a feature

Reply:
made it internal and renamed it, its called when the feature is installed